websocket/server_test.go

```diff
@@ -95,64 +114,163 @@ func TestSubscription(t *testing.T) {
 		PongTimeout:      5 * time.Second,
 	}
 
-	wsServer := NewServer(httpServer, "/wstest", shttp.NewNoAuthenticationBackend(), serverOpts)
+	wsServer := NewServer(httpServer, "/"+path, shttp.NewNoAuthenticationBackend(), serverOpts)
 
-	serverHandler := &fakeServerSubscriptionHandler{t: t, server: wsServer, connected: 0, received: 0}
-	wsServer.AddEventHandler(serverHandler)
+	handler := &fakeServerSubscriptionHandler{t: s.t, server: wsServer}
+	wsServer.AddEventHandler(handler)
+	s.handler = handler
 
 	wsServer.Start()
-	defer wsServer.Stop()
+	s.wsServer = wsServer
+}
+
+func (s *testServer) stop() {
+	if s.wsServer != nil {
+		s.wsServer.Stop()
+	}
+	if s.httpServer != nil {
+		s.httpServer.Stop()
+	}
+}
+
+func (s *testServer) test(numClients int) error {
+	// serverHandler should be notified once:
+	// - by server (Server)
+	if s.handler.connected != 1*numClients {
+		return fmt.Errorf("Server should have received 1 OnConnected event: %v", s.handler.connected)
```

Comment:
return fmt.Errorf("Server should have received %d OnConnected event: %v", numClients, s.handler.connected)

Reply:
fixed